ugs-core/src/com/willwinder/universalgcodesender/gcode/util/GcodeUtils.java

```diff
@@ -48,20 +53,22 @@ public static String unitCommand(Units units) {
      * Generates a move command given a base command. The command will be appended with the relative movement to be made
      * on the axises with the given distance and feed rate.
      *
-     * @param command  the base command to use, ie: G20G91G1 or G1
+     * @param command  the base command to use, ie: G91G1 or G1
      * @param distance the distance to move in the currently selected unit (G20 or G21)
      * @param dirX     1 for positive movement, 0 for no movement, -1 for negative movement
      * @param dirY     1 for positive movement, 0 for no movement, -1 for negative movement
      * @param dirZ     1 for positive movement, 0 for no movement, -1 for negative movement
+     * @param units    the units to use for movement
      */
-    public static String generateMoveCommand(String command, double distance, double feedRate, int dirX, int dirY, int dirZ) {
+    public static String generateMoveCommand(String command, double distance, double feedRate, int dirX, int dirY, int dirZ, Units units) {
```

Comment:
If this returned a list of commands would TinyG still need custom code to handle the movements?